cluster/src/main/java/com/hartwig/pipeline/metadata/BiopsyMetadataApi.java

```diff
@@ -62,6 +85,49 @@ public void start() {
 
     @Override
     public void complete(final PipelineState state, final SomaticRunMetadata metadata) {
-        // do nothing
+        if (state.status() != PipelineStatus.FAILED) {
+            List<AddDatatype> addDatatypes =
+                    state.stageOutputs().stream().map(StageOutput::datatypes).flatMap(List::stream).collect(Collectors.toList());
+            SampleSet set = OnlyOne.of(setApi.list(metadata.set(), null), SampleSet.class);
+            ImmutablePipelineStaged.Builder stagedEventBuilder = ImmutablePipelineStaged.builder()
+                    .type(PipelineStaged.Type.DNA)
+                    .version(Versions.pipelineMajorMinorVersion())
+                    .sample(metadata.tumor().sampleName())
+                    .setId(set.getId());
+            OutputIterator.from(blob -> {
+                Optional<DataType> dataType =
+                        addDatatypes.stream().filter(d -> blob.getName().endsWith(d.path())).map(AddDatatype::dataType).findFirst();
```

Comment:
Should we include `/` in the `endsWith` string as an anchor? eg `...endsWith("/" + d.path()))...`?